src/Sprite/OP2BmpLoader.cpp

```diff
@@ -5,17 +5,14 @@
 #include <algorithm>
 #include <limits>
 
-OP2BmpLoader::OP2BmpLoader(std::string bmpFilename, std::string artFilename) :
-	bmpReader(bmpFilename), artFile(ArtFile::Read(artFilename)) { }
-
-OP2BmpLoader::OP2BmpLoader(std::string bmpFilename, Stream::BidirectionalReader& artFileStream) :
-	bmpReader(bmpFilename), artFile(ArtFile::Read(artFileStream)) { }
+OP2BmpLoader::OP2BmpLoader(std::string bmpFilename, std::shared_ptr<ArtFile> artFile) :
+	bmpReader(bmpFilename), artFile(artFile) { }
 
 void OP2BmpLoader::ExtractImage(std::size_t index, const std::string& filenameOut) 
 {
-	artFile.VerifyImageIndexInBounds(index);
+	artFile->VerifyImageIndexInBounds(index);
 
-	ImageMeta& imageMeta = artFile.imageMetas[index];
+	ImageMeta& imageMeta = artFile->imageMetas[index];
```

Comment:
Ahh, I guess storing references to existing data is why you used a `std::shared_ptr`.